src/main/scala/org/scalajs/dom/HTMLCollection.scala

```diff
@@ -14,7 +14,7 @@ import scala.scalajs.js.annotation._
   */
 @js.native
 @JSGlobal
-class HTMLCollection private[this] () extends DOMList[Element] {
+abstract class HTMLCollection extends DOMList[Element] {
```

Comment:
Consider keeping the constructor package-private, as I don't think it's possible to extend this class in user code:
```suggestion
abstract class HTMLCollection private[dom] extends DOMList[Element] {
```

Reply:
Also, can we use a type parameter? e.g. 
```suggestion
abstract class HTMLCollection[E <: Element] extends DOMList[E] {
```

This way, you wouldn't have to override `item` and `namedItem` to get more specific return types.